src/deferred.js

```diff
@@ -103,7 +103,10 @@ jQuery.extend( {
 						fns = null;
 					} ).promise();
 				},
-				then: function( onFulfilled, onRejected, onProgress ) {
+
+                // The onException is there to manage that an exception is thrown
+                // inside the onFullfilled without being catched.
+				then: function( onFulfilled, onRejected, onProgress, onException ) {
```

Comment:
OK, will move to that direction. My concern is though that if we want to provide `$( fn, except )` in API, then public selector call must be modified in `init.js`, as I have done. Should I do it another way?